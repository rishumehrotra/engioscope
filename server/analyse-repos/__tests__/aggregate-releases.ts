import { EnvironmentStatus } from 'azure-devops-node-api/interfaces/ReleaseInterfaces';
import aggregateReleases from '../aggregate-releases';

test('it should aggregate releases (single release)', () => {
  const byRepoId = aggregateReleases([
    {
      artifacts: [{
        definitionReference: {
          repository: { id: 'repo-id-1', name: 'some-repo-1' },
          branch: { id: 'refs/heads/master', name: 'refs/heads/master' }
        }
      }],
      environments: [{
        name: 'SIT',
        status: EnvironmentStatus.Succeeded,
        deploySteps: [{ lastModifiedOn: new Date('2020-10-10T10:10') }]
      }]
    }
  ]);

  expect(byRepoId('repo-id-1')).toMatchSnapshot();
  expect(byRepoId('something-else')).toMatchSnapshot();
});

test('it should aggregate releases (two releases, different environments)', () => {
  const byRepoId = aggregateReleases([
    {
      artifacts: [{
        definitionReference: {
          repository: { id: 'repo-id-1', name: 'some-repo-1' },
          branch: { id: 'refs/heads/master', name: 'refs/heads/master' }
        }
      }],
      environments: [{
        name: 'SIT',
        status: EnvironmentStatus.Succeeded,
        deploySteps: [{ lastModifiedOn: new Date('2020-10-10T10:10') }]
      }]
    },
    {
      artifacts: [{
        definitionReference: {
          repository: { id: 'repo-id-1', name: 'some-repo-1' },
          branch: { id: 'refs/heads/master', name: 'refs/heads/master' }
        }
      }],
      environments: [{
        name: 'Preprod',
        status: EnvironmentStatus.Succeeded,
        deploySteps: [{ lastModifiedOn: new Date('2020-10-10T10:10') }]
      }]
    }
  ]);

  expect(byRepoId('repo-id-1')).toMatchSnapshot();
});

test('it should aggregate releases (mutiple releases, repeating environments)', () => {
  const byRepoId = aggregateReleases([
    {
      artifacts: [{
        definitionReference: {
          repository: { id: 'repo-id-1', name: 'some-repo-1' },
          branch: { id: 'refs/heads/master', name: 'refs/heads/master' }
        }
      }],
      environments: [{
        name: 'SIT',
        status: EnvironmentStatus.Succeeded,
        deploySteps: [{ lastModifiedOn: new Date('2020-10-10T10:10') }]
      }, {
        name: 'preprod',
        status: EnvironmentStatus.Succeeded,
        deploySteps: [{ lastModifiedOn: new Date('2020-10-10T10:10') }]
      }]
    },
    {
      artifacts: [{
        definitionReference: {
          repository: { id: 'repo-id-1', name: 'some-repo-1' },
          branch: { id: 'refs/heads/master', name: 'refs/heads/master' }
        }
      }],
      environments: [{
        name: 'SIT',
        status: EnvironmentStatus.Succeeded,
        deploySteps: [{ lastModifiedOn: new Date('2020-10-10T10:10') }]
      }, {
        name: 'preprod',
        status: EnvironmentStatus.Succeeded,
        deploySteps: [{ lastModifiedOn: new Date('2020-10-10T10:10') }]
      }]
    }
  ]);

  expect(byRepoId('repo-id-1')).toMatchSnapshot();
});
