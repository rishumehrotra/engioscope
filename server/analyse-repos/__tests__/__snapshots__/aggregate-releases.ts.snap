// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should aggregate releases (mutiple releases, repeating environments) 1`] = `
Object {
  "count": 4,
  "indicators": Array [
    Object {
      "name": "Last deployment date (SIT)",
      "rating": 0,
      "value": "Oct 10",
    },
    Object {
      "name": "Total deployments (SIT)",
      "rating": 4,
      "value": 2,
    },
    Object {
      "name": "Successful deployment rate (SIT)",
      "rating": 100,
      "value": "100%",
    },
    Object {
      "name": "Non master deployments (SIT)",
      "rating": 100,
      "value": 0,
    },
    Object {
      "name": "Last deployment date (PP)",
      "rating": 0,
      "value": "Oct 10",
    },
    Object {
      "name": "Total deployments (PP)",
      "rating": 4,
      "value": 2,
    },
    Object {
      "name": "Successful deployment rate (PP)",
      "rating": 100,
      "value": "100%",
    },
    Object {
      "name": "Non master deployments (PP)",
      "rating": 100,
      "value": 0,
    },
    Object {
      "name": "Last deployment date (PROD)",
      "rating": 0,
      "value": "never",
    },
    Object {
      "name": "Total deployments (PROD)",
      "rating": 0,
      "value": 0,
    },
    Object {
      "name": "Successful deployment rate (PROD)",
      "rating": 0,
      "value": "0%",
    },
    Object {
      "name": "Non master deployments (PROD)",
      "rating": 100,
      "value": 0,
    },
  ],
  "name": "Releases",
  "rating": 43,
}
`;

exports[`it should aggregate releases (single release) 1`] = `
Object {
  "count": 1,
  "indicators": Array [
    Object {
      "name": "Last deployment date (SIT)",
      "rating": 0,
      "value": "Oct 10",
    },
    Object {
      "name": "Total deployments (SIT)",
      "rating": 2,
      "value": 1,
    },
    Object {
      "name": "Successful deployment rate (SIT)",
      "rating": 100,
      "value": "100%",
    },
    Object {
      "name": "Non master deployments (SIT)",
      "rating": 100,
      "value": 0,
    },
    Object {
      "name": "Last deployment date (PP)",
      "rating": 0,
      "value": "never",
    },
    Object {
      "name": "Total deployments (PP)",
      "rating": 0,
      "value": 0,
    },
    Object {
      "name": "Successful deployment rate (PP)",
      "rating": 0,
      "value": "0%",
    },
    Object {
      "name": "Non master deployments (PP)",
      "rating": 100,
      "value": 0,
    },
    Object {
      "name": "Last deployment date (PROD)",
      "rating": 0,
      "value": "never",
    },
    Object {
      "name": "Total deployments (PROD)",
      "rating": 0,
      "value": 0,
    },
    Object {
      "name": "Successful deployment rate (PROD)",
      "rating": 0,
      "value": "0%",
    },
    Object {
      "name": "Non master deployments (PROD)",
      "rating": 100,
      "value": 0,
    },
  ],
  "name": "Releases",
  "rating": 34,
}
`;

exports[`it should aggregate releases (single release) 2`] = `
Object {
  "count": 0,
  "indicators": Array [
    Object {
      "name": "Last deployment date (SIT)",
      "rating": 0,
      "value": "never",
    },
    Object {
      "name": "Total deployments (SIT)",
      "rating": 0,
      "value": 0,
    },
    Object {
      "name": "Successful deployment rate (SIT)",
      "rating": 0,
      "value": "0%",
    },
    Object {
      "name": "Non master deployments (SIT)",
      "rating": 100,
      "value": 0,
    },
    Object {
      "name": "Last deployment date (PP)",
      "rating": 0,
      "value": "never",
    },
    Object {
      "name": "Total deployments (PP)",
      "rating": 0,
      "value": 0,
    },
    Object {
      "name": "Successful deployment rate (PP)",
      "rating": 0,
      "value": "0%",
    },
    Object {
      "name": "Non master deployments (PP)",
      "rating": 100,
      "value": 0,
    },
    Object {
      "name": "Last deployment date (PROD)",
      "rating": 0,
      "value": "never",
    },
    Object {
      "name": "Total deployments (PROD)",
      "rating": 0,
      "value": 0,
    },
    Object {
      "name": "Successful deployment rate (PROD)",
      "rating": 0,
      "value": "0%",
    },
    Object {
      "name": "Non master deployments (PROD)",
      "rating": 100,
      "value": 0,
    },
  ],
  "name": "Releases",
  "rating": 25,
}
`;

exports[`it should aggregate releases (two releases, different environments) 1`] = `
Object {
  "count": 2,
  "indicators": Array [
    Object {
      "name": "Last deployment date (SIT)",
      "rating": 0,
      "value": "Oct 10",
    },
    Object {
      "name": "Total deployments (SIT)",
      "rating": 2,
      "value": 1,
    },
    Object {
      "name": "Successful deployment rate (SIT)",
      "rating": 100,
      "value": "100%",
    },
    Object {
      "name": "Non master deployments (SIT)",
      "rating": 100,
      "value": 0,
    },
    Object {
      "name": "Last deployment date (PP)",
      "rating": 0,
      "value": "Oct 10",
    },
    Object {
      "name": "Total deployments (PP)",
      "rating": 2,
      "value": 1,
    },
    Object {
      "name": "Successful deployment rate (PP)",
      "rating": 100,
      "value": "100%",
    },
    Object {
      "name": "Non master deployments (PP)",
      "rating": 100,
      "value": 0,
    },
    Object {
      "name": "Last deployment date (PROD)",
      "rating": 0,
      "value": "never",
    },
    Object {
      "name": "Total deployments (PROD)",
      "rating": 0,
      "value": 0,
    },
    Object {
      "name": "Successful deployment rate (PROD)",
      "rating": 0,
      "value": "0%",
    },
    Object {
      "name": "Non master deployments (PROD)",
      "rating": 100,
      "value": 0,
    },
  ],
  "name": "Releases",
  "rating": 42,
}
`;
